airflow/providers/http/hooks/http.py

```diff
@@ -363,8 +360,10 @@ async def run(
             else:
                 raise AirflowException(f"Unexpected HTTP Method: {self.method}")
 
-            attempt_num = 1
-            while True:
+            for attempt in range(1, 1 + self.retry_limit):
+                if attempt > 1:
+                    await asyncio.sleep(self.retry_delay)
```

Comment:
Would it not work if the `sleep()` is called at the end of the loop instead? This does a `return` when the condition is satisfied and would be able to achieve the same without the `if` condition. Am I missing something?

Reply:
I was actually able to put it into the `except:` block.